operator/pkg/certs/certs.go

```diff
@@ -247,14 +246,16 @@ func (cert *KarmadaCert) KeyName() string {
 	return pair + keyExtension
 }
 
-// GeneratePrivateKey generates cert key with default size if 1024. it supports
-// ECDSA and RAS algorithm.
+// GeneratePrivateKey generates a certificate key. It supports both
+// ECDSA (using the P-256 elliptic curve) and RSA algorithms. For RSA,
+// the key is generated with a size of 3072 bits.
 func GeneratePrivateKey(keyType x509.PublicKeyAlgorithm) (crypto.Signer, error) {
 	if keyType == x509.ECDSA {
 		return ecdsa.GenerateKey(elliptic.P256(), cryptorand.Reader)
+	} else if keyType == x509.RSA {
```

Comment:
https://github.com/karmada-io/karmada/blob/c230b14827d01c09db98b9531e4224d93429ee67/pkg/karmadactl/cmdinit/cert/cert.go#L58-L64

`karmadactl init` has a similar implementation; can this be changed together?